pallets/parachain-staking/src/lib.rs

```diff
@@ -2388,28 +2437,94 @@ pub mod pallet {
 					imb.peek(),
 				));
 			}
+
+			let payout = DelayedPayout {
+				round_issuance: total_issuance,
+				total_staking_reward: left_issuance,
+				collator_commission: <CollatorCommission<T>>::get(),
+			};
+
+			<DelayedPayouts<T>>::insert(
+				round_to_payout,
+				payout
+			);
+		}
+
+		/// Wrapper around pay_one_collator_reward which handles the following logic:
+		/// * whether or not a payout needs to be made
+		/// * cleaning up when payouts are done
+		/// * returns whether or not a payout was made
+		///
+		/// TODO: settle on (and document) return value
+		fn handle_delayed_payouts(now: RoundIndex) -> bool {
+			// cases:
+			// 1. payouts doesn't exist: nothing to do
+			//		caveat/TODO: on first round after this upgrade, this will be true and we won't
+			//		clean up? a migration could solve this...
+			// 2. payouts exists and we aren't done
+			// 3. payouts exists and we are done (and we should clean up)
+
+            // we call the current round the "payout_round" and the round we are paying out for the
+            // "paid_for_round".
+            // payout_round is paid_for_round + RewardPaymentDelay + 1 and starts at the first
+            // block.
+
+			let delay = T::RewardPaymentDelay::get();
+            // don't underflow uint
+            if now < delay {
+                return false;
+            }
+
+            let paid_for_round = now - delay;
+
+			if let Some(payout_info) = <DelayedPayouts<T>>::get(paid_for_round) {
+				if let None = Self::pay_one_collator_reward(paid_for_round, payout_info) {
```

Comment:
```suggestion
				// This call returns none when all collators who were due payment have been paid.
				if let None = Self::pay_one_collator_reward(paid_for_round, payout_info) {
```

Reply:
The most recent code includes some comments about this, but feel free to add more clarity